File: 56-merge-intervals/merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals = sorted(intervals)
        ans = [intervals[0]]
        for s, e in intervals:
            if s <= ans[-1][1]:
                ans[-1][1] = max(ans[-1][1], e)
            else:
                ans.append([s,e])
        return ans

        

File: 56-merge-intervals/merge-intervals.go
import "sort"

func merge(intervals [][]int) [][]int {

    res:= make([][]int,0 )

   
    sort.Slice(intervals, func(i,j int) bool {
        return intervals[i][0]<=intervals[j][0]
    })
    res = append(res, intervals[0])

    for i:=0; i<len(intervals);i++{
        start:= intervals[i][0]
        end:= intervals[i][1]
        l:=len(res)-1
        if start <= res[l][1] {
            res[l][1]= max(end, res[l][1])
        } else {
            res= append(res, []int{start, end})
        }
    }
    return res
    
}